components/server/ee/src/gitlab/gitlab-app-support.ts

```diff
@@ -9,6 +9,20 @@ import { inject, injectable } from "inversify";
 import { TokenProvider } from "../../../src/user/token-provider";
 import { UserDB } from "@gitpod/gitpod-db/lib";
 import { Gitlab } from "@gitbeaker/node";
+import { ProjectSchemaDefault, NamespaceInfoSchemaDefault } from "@gitbeaker/core/dist/types/services/Projects";
+
+// Add missing fields to Gitbeaker's ProjectSchema type
+type ProjectSchema = ProjectSchemaDefault & {
+    last_activity_at: string;
+    namespace: NamespaceInfoSchemaDefault & {
+        avatar_url: string;
```

Comment:
Good catch! Found [a way to support emojis](https://dev.to/acanimal/how-to-slice-or-get-symbols-from-a-unicode-string-with-emojis-in-javascript-lets-learn-how-javascript-represent-strings-h3a) as first character. 👍

Reply:
@gtsiolis Could you please try again, or invite me to that group? (For some reason, I'm not able to create one that starts with an emoji)

Reply:
@jankeromnes It almost worked! I added this to the reproduction-group in GitLab (see [example project](https://gitlab.com/reproduction-group/sample-group/sample-project)), but somehow the emoji is surfacing to the top-level group. Previously, refrained from adding this as it would break new project wizard for you, too. 🤷  

<img width="487" alt="Screenshot 2022-03-30 at 7 01 12 PM" src="https://user-images.githubusercontent.com/120486/160901253-ee7cebe1-4aa5-4666-b1c6-41f509db2a09.png">

Reply:
@gtsiolis Ah, I think I got it:
- Previously, we picked the avatar only for the `owner` (when provided by GitLab)
- With this Pull Request, we also look at the `namespace` when the `owner` is not available
- However, I see that `namespace` is not always the "root namespace" (i.e. for `reproduction-group / 🍊-sample-group / 🍋-sample-project`, GitLab provides `🍊-sample-group`, not `reproduction-group`)

I believe the fix is to only use the avatar for `owner` if provided, or `namespace` if it is the root namespace. (Note: We could also look at the `parent_id` of nested groups, then make follow-up GitLab API requests to find the root namespace and its avatar -- but I think that would be way over-engineered at this point.)

Let's see how the new algorithm behaves. Pushing soon.

Reply:
Looks resolved now. Thanks for fixing this, @jankeromnes!